test/tests/functional/pbs_reservations.py

```diff
@@ -85,56 +86,324 @@ def submit_asap_reservation(self, user, jid):
+        a = {'reserve_retry_time': 5}
         self.server.manager(MGR_CMD_SET, SERVER, a)
 
-        a = {'resources_available.ncpus': 4}
+        a = {'resources_available.ncpus': 1}
         self.server.create_vnodes('vn', a, num=2, mom=self.mom)
 
         now = int(time.time())
 
         # submitting 25 seconds from now to allow some of the older testbed
         # systems time to process (discovered empirically)
-        rid = self.submit_standing_reservation(user=TEST_USER,
-                                               select='1:ncpus=4',
-                                               rrule='FREQ=HOURLY;COUNT=3',
-                                               start=now + 25,
-                                               end=now + 45)
+        rid = self.submit_reservation(user=TEST_USER, select='1:ncpus=1',
+                                      rrule=rrule, start=start, end=end)
 
         a = {'reserve_state': (MATCH_RE, 'RESV_CONFIRMED|2')}
         self.server.expect(RESV, a, id=rid)
 
         self.server.status(RESV, 'resv_nodes', id=rid)
         resv_node = self.server.reservations[rid].get_vnodes()[0]
 
-        a = {'reserve_state': (MATCH_RE, 'RESV_RUNNING|5')}
-        offset = 25 - (int(time.time()) - now)
-        self.server.expect(RESV, a, id=rid, offset=offset, interval=1)
+        if run:
+            resv_state = {'reserve_state': (MATCH_RE, 'RESV_RUNNING|5')}
+            self.logger.info('Sleeping until reservation starts')
+            offset = start - int(time.time())
+            self.server.expect(RESV, resv_state, id=rid,
+                               offset=offset, interval=1)
+        else:
+            resv_state = {'reserve_state': (MATCH_RE, 'RESV_DEGRADED|10')}
 
         a = {'state': 'offline'}
         self.server.manager(MGR_CMD_SET, NODE, a, id=resv_node)
 
-        a = {'reserve_state': (MATCH_RE, 'RESV_RUNNING|5'),
-             'reserve_substate': 10}
-        self.server.expect(RESV, a, attrop=PTL_AND, id=rid)
+        a = {'reserve_substate': 10}
+        a.update(resv_state)
+        self.server.expect(RESV, a, id=rid)
 
         a = {'resources_available.ncpus': (GT, 0)}
         free_nodes = self.server.filter(NODE, a)
         nodes = list(free_nodes.values())[0]
 
         other_node = [nodes[0], nodes[1]][resv_node == nodes[0]]
 
-        a = {'reserve_state': (MATCH_RE, 'RESV_CONFIRMED|2'),
-             'resv_nodes': (MATCH_RE, re.escape(other_node))}
-        offset = 45 - (int(time.time()) - now)
-        self.server.expect(RESV, a, id=rid, interval=1, offset=offset,
-                           attrop=PTL_AND)
+        if run:
+            a = {'reserve_substate': 5}
+        else:
+            a = {'reserve_substate': 2}
+        a.update({'resv_nodes': (MATCH_RE, re.escape(other_node))})
+
+        if not run:
+            resv_state = {'reserve_state': (MATCH_RE, 'RESV_CONFIRMED|2')}
+        self.server.expect(RESV, a, id=rid, interval=1)
```

Comment:
I think we should set trigger_sched_cycle option as false, just to make sure that the cycle is triggered based on the reserve_retry_time and not because of expect triggering it,

Reply:
It won't matter if it is based on reserve_retry_time or not.  The scheduler will ignore a reservation whose retry time is not up.  There really is no reason to run cycles through.